### PR TITLE
feat: using view transition api to flip pages

### PR DESCRIPTION
**Pull request by zoolsher**
## Summary

This PR provides a demo of using the View Transition API for screen transitions, but it has not yet added the switch functionality (runtime or compile-time).

This is still WIP:

- [x] Find an new elegant animation by default
- [x] API for custom animation
- [x] Support browserHistory
- [x] How to work with forward and back button of browser. Maybe `popstate` event?

## Details

https://github.com/web-infra-dev/rspress/assets/9161085/132b4668-deab-4b53-ac1f-eacb404b294c

## Checklist

- [x] I have added changeset via `pnpm run change`.
- [x] I have updated the documentation.
- [x] I have added tests to cover my changes.

### CODE REVIEW

**Comment by zoolsher** [comment]
@sanyuan0704 Do you have any suggestions on how to test this animation?

**Comment by sanyuan0704** [comment]
I found it's hard to test the animation in e2e case

File: packages/document/docs/zh/api/config/config-theme.mdx [hunk 1]
```diff
@@ -447,3 +447,23 @@ export default defineConfig({
...
+
+
+## enableContentAnimation
+
+- Type: `boolean`
+- Default: `false`
+
+在页面切换的时候是否显示转场动画，例如：
+
+> 转场动画暂时不能配置
```
Comment (sanyuan0704) [inline packages/document/docs/zh/api/config/config-theme.mdx:459]: There should be a period in the end.

Comment (zoolsher) [reply packages/document/docs/zh/api/config/config-theme.mdx]: fixed

**Comment by sanyuan0704** [comment]
There is a possential bug in e2e test. If i switched the `enableContentAnimation` to `false`, the test process will block all the time.

**Review by sanyuan0704 — APPROVED** [review]

**Comment by sanyuan0704** [comment]
🎉 Awesome!

**Pull request merged**